apps/definition/lib/definition/schema/type.ex

```diff
@@ -48,4 +48,14 @@ defmodule Definition.Schema.Type do
   def access_path do
     one_of([spec(is_binary()), coll_of(spec(is_binary()))])
   end
+
+  @spec impl_of(module) :: spec
+  def impl_of(protocol) do
```

Comment:
If I understand this correctly (and I'm not sure I do), I could see this being really handy.

Reply:
checking if something is an implementation for a protocol, or what protocols have been implemented isn't anything new, but this would allow us to enforce Norm validation that the correct protocol has been implemented.